### ISSUE TITLE
vim-illuminate has higer priority than * highlighting

### ISSUE THREAD

**Issue by andreyorst** [RRethy/vim-illuminate#13]
I did encountered this with my own implementation (yes, [it's me](https://www.reddit.com/r/vim/comments/92jvpa/vimilluminate_selectively_illuminate_other_uses/e369cmw)), and didn't find a way to workaround this, though I didn't searched well enough. Maybe you'll came with something up.

When current word is highlighted by illuminate, pressing <kbd>*</kbd> will not result into highlighting work with hlsearch group until another word steals highlighting from illuminate.

**Comment by RRethy** [comment]
Hey again 😄 . So it originally did respect * highlighting but that was using a hacky check:
```
if l:matched_word !~ @/ || !&hls || !v:hlsearch
```
But with your comment I tried it again and now it's no longer respecting * so I think the check above didn't work before and the highlighting was being respected for some other reason (maybe a lucky bug lol).

That being said, I fixed it and now it respects * highlighting. I'm using matchadd() for the highlighting (I think you were using :match if I recall correctly), and matchadd() takes an optional priority value, the lower the value the lower the priority (originally I had it as 10 since I read this the other way around LOL). Hl-search has a priority of 0, so by passing in -1, illuminate will respect *.

[PR for reference](https://github.com/RRethy/vim-illuminate/pull/14)
